### ISSUE TITLE
Media & Text Block: Invalid content for blocks made before WP 5.4

### ISSUE THREAD

**Issue by sandrab** [WordPress/gutenberg#29109]
## Description
When opening the editor for a post that contains a Media & Text block that was made in WP core 5.3 or earlier and was left unchanged since, the migration to the new version of the block does not work anymore in WP 5.6. The editor shows the message 'This block contains unexpected or invalid content.' with the 'Attempt block recovery' button.

## Step-by-step reproduction instructions
1. In WP 5.3 create a post and add a Media & Text block. Don't change any settings - this means the 'Stack on mobile' is false, which was the default at the time.
2. Save the post and close the editor screen.
3. Update to WP 5.6.
4. Open the post again.

## Expected behaviour
The Media & Text block code gets updated to the new version (visible in the console). This adds the missing `isStackedOnMobile` attribute, for example:
`<!-- wp:media-text {"mediaId":6,"mediaType":"image","isStackedOnMobile":false,"className":"alignwide"} -->`

## Actual behaviour
The Media & Text block is shown as invalid.
The console explains why, for example:
```
Block validation: Expected attribute `class` of value `wp-block-media-text is-stacked-on-mobile alignwide`, saw `wp-block-media-text alignwide`.

Block validation: Block validation failed for `core/media-text` ( 
Object { name: "core/media-text", icon: {…}, keywords: (2) […], providesContext: {}, usesContext: [], styles: [], category: "media", title: "Media & Text", description: "Set media and words side-by-side for a richer layout.", example: {…}, … }
 ).
Content generated by `save` function:
<div class="wp-block-media-text is-stacked-on-mobile alignwide"><figure class="wp-block-media-text__media"><img src="http://dev.wp-20210218.org/wp-content/uploads/2021/02/testing-gh.jpg" alt=""/></figure><div class="wp-block-media-text__content"></div></div>
Content retrieved from post body:
<div class="wp-block-media-text alignwide"><figure class="wp-block-media-text__media"><img src="http://dev.wp-20210218.org/wp-content/uploads/2021/02/testing-gh.jpg" alt="" class="wp-image-6"/></figure><div class="wp-block-media-text__content"></div></div>
```

## WordPress information
- WordPress version: 5.6
- Gutenberg version: Not installed
- Are all plugins except Gutenberg deactivated? Yes
- Are you using a default theme (e.g. Twenty Twenty-One)? Yes

## Device information
- Device: Desktop
- Operating system: Windows 10
- Browser: Firefox 85.0.2

**Comment by sandrab** [comment]
My guess is that this behaviour is introduced by the fix for https://github.com/WordPress/gutenberg/issues/26039, where the default value of the isStackedOnMobile attribute was changed from false to true in the block deprecation (in packages/block-library/src/media-text/deprecated.js).

**Comment by getsource** [comment]
I was able to reproduce this.

The default media and text block content from 5.3.8 generates the error, as does one with basic contents (image+text, no other modifications) added.

Here's the default block contents for testing:
```
<!-- wp:media-text -->
<div class="wp-block-media-text alignwide"><figure class="wp-block-media-text__media"></figure><div class="wp-block-media-text__content"><!-- wp:paragraph {"placeholder":"Content…","fontSize":"large"} -->
<p class="has-large-font-size"></p>
<!-- /wp:paragraph --></div></div>
<!-- /wp:media-text -->
```

**Comment by t-hamano** [comment]
@sandrab

Sorry for the delay in contacting you. This issue should be fixed by #49538.
